mlflow/gateway/gateway_app.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+from fastapi import FastAPI, Request, HTTPException
+import logging
+import os
+from traceback import format_exc
+from typing import List
+import uvicorn
+
+from mlflow.version import VERSION
+from mlflow.gateway.constants import CONF_PATH_ENV_VAR
+from mlflow.gateway.handlers import Route, RouteConfig, _route_config_to_route, _load_route_config
+
+
+_logger = logging.getLogger(__name__)
+
+# Configured and initialized Gateway Routes state index
+ACTIVE_ROUTES: List[Route] = []
+
+app = FastAPI(
+    title="MLflow Model Gateway API",
+    description="The core gateway API for reverse proxy interface using remote inference "
+    "endpoints within MLflow",
+    version=VERSION,
+)
+
+
+@app.get("/health")
+async def health():
+    return {"status": "OK"}
+
+
+@app.get("/gateway/routes/{route_name}")
```

Comment:
How does fastapi handle URL encoding? E.g. if my route name has a space in it, does that get resolved properly here (i.e. %20 is decoded)?

Reply:
We don't permit spaces in the endpoints. The function `is_valid_endpoint_name` in `mlflow.gateway.utils` will cause an Exception to be thrown for non-allowed characters in the route name. 
Did you want to support spaces?